tests/unit/restapi/v1/test_experiments.py

```diff
@@ -0,0 +1,859 @@
+    expected: dict[str, Any],
+) -> None:
+    """Assert that an experiment's associated entrypoints can be updated for an
+    experiment with specified id.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to add entrypoints.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    payload = {"entrypoints": entrypoints}
+    response = client.post(
+        f"/{V1_ROOT}/{V1_EXPERIMENTS_ROUTE}/{experiment_id}",
+        json=payload,
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json == expected
+
+
+def assert_removing_experiment_entrypoints_works(
+    client: FlaskClient,
+    experiment_id: int,
+    entrypoints_to_remove: list[int],
+    expected: dict[str, Any],
+) -> None:
+    """Assert that an experiment's associated entrypoints can be removed for an
+    experiment with specified id.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to replace entrypoints.
+        entrypoints: The entrypoints to remove from the experiment.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    current_experiment = client.get(
+        f"/{V1_ROOT}/{V1_EXPERIMENTS_ROUTE}/{experiment_id}",
+        follow_redirects=True,
+    )
+    experiment_json = current_experiment.get_json()
+
+    updated_entrypoints = []
+    for entry in experiment_json["entrypoints"]:
+        if entry not in entrypoints_to_remove:
+            updated_entrypoints.append(entry)
+
+    payload = {
+        "entrypoints": updated_entrypoints,
+        "name": experiment_json["name"],
+        "group_id": experiment_json["group_id"],
+    }
+    response = client.put(
+        f"/{V1_ROOT}/{V1_EXPERIMENTS_ROUTE}/{experiment_id}",
+        json=payload,
+        follow_redirects=True,
+    )
+
+    assert response.status_code == 200 and response.json == expected
+
+
+def assert_replacing_experiment_entrypoints_works(
+    client: FlaskClient,
+    experiment_id: int,
+    entrypoints: list[int],
+    expected: dict[str, Any],
+) -> None:
+    """Assert that an experiment's associated entrypoints can be updated for
+    an experiment with specified id.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to replace entrypoints.
+        entrypoints: The new entrypoints for the experiment.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    current_experiment = client.get(
```

Comment:
We should pass one of the experiments from the fixture from the test function instead of issuing a GET.